src/ui.rs

```diff
@@ -25,18 +31,28 @@ pub fn default_view<B: Backend>(f: &mut Frame<B>) {
         .constraints([Constraint::Percentage(70), Constraint::Percentage(30)].as_ref())
         .split(chunks[0]);
 
-    tasks_view(f, top_chunks[0]);
+    tasks_view(f, top_chunks[0], &mut app.state);
     calendar_view(f, top_chunks[1]);
-    footer_view(f, chunks[1]);
+    footer_view(f, chunks[1], app);
 }
 
-fn tasks_view<B: Backend>(f: &mut Frame<B>, area: Rect) {
+fn tasks_view<B: Backend>(f: &mut Frame<B>, area: Rect, state: &mut State) {
```

Comment:
I dont think that UI should accept mutable reference to State, it SHOULD NOT modify any data. Readonly reference will suffice i think

Reply:
render_stateful_widget expects &mut State (ListState in this case), which is currently held in main State stuct, It will be resolved if we come up with new solution for concern in state.rs

Reply:
right, i missed that you changed draw call method to _render_stateful_widget_, For now lets leave it like that but i propose to come up with a different solution